bin/build.sh

```diff
@@ -1,73 +1,81 @@
   find "build/$1" -name '*.md' | while read file; do
-    echo "Processing $file using the ${1} stylesheet"
+    echo "Converting ${file#build/}..."
 
-    pdf_name="${file%.md}.pdf"
-    html_name="${file%.md}.html"
-    
-    file_headline=$(head -n 1 "$file")
-    document_title=$(echo "$file_headline" | sed -E "s/#+\s*//")
+    pdf_path="${file%.md}.pdf"
+    html_path="${file%.md}.html"
+    # Use first header as the document title; remove all # characters and whitespace from the beginning
+    document_title=$(head -n 1 "$file" | sed -E "s/^#+\s*//")
+    # Absolute path to the custom stylesheet
+    custom_stylesheet=$(realpath "./assets/$1-style.css")
 
     # Replace wca{...} and wcadoc{...} with absolute WCA URLs (the correct URLs only get inserted in production)
     sed -Ei "s#wca\{([^}]*)\}#$wca_url\1#g" "$file"
     sed -Ei "s#wcadoc\{([^}]*)\}#$wca_docs_url\1#g" "$file"
     # Replace {logo} with the path to the WCA logo in /assets
     sed -Ei "s#\{logo\}#$logo#g" "$file"
 
-    # This creates the custom header for each edudoc and saves it next to the doc that will use it
-    if [ $1 = "edudoc" ]; then
-      header_html="${file%.md}-header.html"
-    
+    # This creates the custom header for each edudoc and prepends it to each file
+    if [ "$1" = "edudoc" ]; then
       # Set the document title, path to the logo and the date
-      sed -E "s#DOCUMENT_TITLE#$document_title#g" "$edudoc_header" |
-      sed -E "s#WCA_LOGO_PATH#$logo#g" |
-      sed -E "s#DATE#$compile_date#g" > "$header_html"
+      sed -E "s#\{document_title\}#$document_title#g" "$edudoc_header" |
+      sed -E "s#\{wca_logo_path\}#$logo#g" |
+      sed -E "s#\{date\}#$compile_date#g" |
+      echo -e "$(cat /dev/stdin)\n\n$(cat $file)" > "$file"
     fi
 
-    # Markdown -> HTML
-    pandoc -s --from markdown --to html5 --css=$2 --metadata pagetitle="$document_title" "$file" -o "$html_name"
+    # Markdown => HTML + remove unsupported styles generated by pandoc
+    pandoc -s --from markdown --to html5 --css "$main_stylesheet" --css "$custom_stylesheet" --metadata pagetitle="$document_title" "$file" |
+    sed -E "s#gap: min\(4vw, 1\.5em\);##" |
+    sed -E "s#overflow-x: auto;##" > "$html_path"
 
     # HTML -> PDF
-    if [ $1 = "documents" ]; then
-      wkhtmltopdf --encoding 'utf-8' -T 15mm -B 15mm -R 15mm -L 15mm --quiet "$html_name" "$pdf_name"
-    elif [ $1 = "edudoc" ]; then
-      wkhtmltopdf --encoding 'utf-8' -T 15mm -B 15mm -R 15mm -L 15mm --header-html "$header_html" --footer-center "[page]" --quiet "$html_name" "$pdf_name"
+    if [ "$1" = "documents" ]; then
+      # --presentational-hints makes things like the start attribute of ol elements work,
+      # which is required for adding more content between two consecutive ordered list items
+      weasyprint --presentational-hints --encoding 'utf-8' "$html_path" "$pdf_path"
+    elif [ "$1" = "edudoc" ]; then
+      weasyprint --presentational-hints --encoding 'utf-8' "$html_path" "$pdf_path"
```

Comment:
The two cases are completely identical as far as I can tell, so there's no need for an if-else.

Reply:
Oops, good catch. There used to be a difference with wkhtmltopdf.